### PR TITLE
Multiple fixes

### PR DESCRIPTION
**Pull request by r-darwish**

### CODE REVIEW

File: src/browser.rs [hunk 1]
```diff
@@ -14,22 +14,27 @@ use crate::process::Process;
 use crate::tab::Tab;
 use crate::transport::Transport;
 
+/// Chrome browser.
 pub struct Browser {
-    pub process: Process,
+    _process: Process,
```
Comment (lukaslueg) [inline src/browser.rs]: An underscore name here unnecessary, just removing `pub` suffices.

File: src/browser.rs [hunk 2]
```diff
@@ -109,12 +114,20 @@ impl Browser {
...
+    /// See the `cdtp` module documentation for available methods.
     pub fn call_method<C>(&self, method: C) -> Result<C::ReturnObject, Error>
     where
         C: cdtp::Method + serde::Serialize,
     {
         self.transport.call_method(method)
     }
+
+    #[cfg(test)]
+    pub(crate) fn process(&self) -> &Process {
```
Comment (atroche) [inline src/browser.rs]: neat, I hadn't thought to do this.

File: src/lib.rs [hunk 3]
```diff
@@ -13,3 +13,5 @@ pub mod tab;
 pub mod transport;
 pub mod waiting_call_registry;
 pub mod web_socket_connection;
+
+pub use browser::{Browser, LaunchOptions};
```
Comment (atroche) [inline src/lib.rs]: Cool.

File: src/process.rs [hunk 4]
```diff
@@ -51,19 +63,13 @@ pub struct LaunchOptions {
...
+        // $BROWSER may also provide default arguments, which we may
+        // or may not override later on.
 
-            // TODO More paths!?
-            let default_paths = &["/usr/bin/google-chrome-stable"][..];
-            for path in default_paths {
-                if std::path::Path::new(path).exists() {
-                    return Some(path.into());
-                }
+        for app in &["google-chrome-stable", "chromium"] {
```
Comment (atroche) [inline src/process.rs]:
```suggestion
        for app in &["google-chrome-stable", "google-chrome", "chromium"] {
```

On my Linux box it's `google-chrome`.

File: src/browser.rs [hunk 5]
```diff
@@ -93,7 +98,7 @@ impl Browser {
                     Event::TargetInfoChanged(ev) => {
                         let target_info = ev.params.target_info;
                         trace!("Target info changed: {:?}", target_info);
-                        if target_info.target_type == "page" {
+                        if target_info.target_type.is_page() {
```
Comment (atroche) [inline src/browser.rs]: How do you feel about this?

```rust
if let TargetType::Page = target_info.target_type {
}
```

I'm happy either way.

**Comment by atroche** [comment]
I'm going to merge this, and then push a commit to change `_process` back to `process`.

**Comment by atroche** [comment]
Oh, scratch that, I see why you did it @r-darwish:

image

File: src/browser.rs (hunk 1 above)
Comment (atroche) [reply src/browser.rs]: https://github.com/atroche/rust-headless-chrome/pull/33#issuecomment-464397840

**Pull request merged**